File: FrontEnd/src/app/classes/task.ts
import {responseUser, uidUser } from './user';

export class Task {
    id: number = 0;
    title: string = '';
    description: string = '';
    status: string = '';
    dueDate: string = '';

    constructor(id: number, title: string, description: string, status: string, dueDate: string) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.status = status;
        this.dueDate = dueDate;
    }
}

export class TaskNoId {
    title: string = '';
    description: string = '';
    status: string = '';
    dueDate: string = '';
    user: responseUser;

    constructor(title: string, description: string, status: string, dueDate: string, user: responseUser) {
        this.title = title;
        this.description = description;
        this.status = status;
        this.dueDate = dueDate;
        this.user = user;
    }
}

File: BackEnd/src/main/java/com/genspark/ToDoList/entity/Task.java
package com.genspark.ToDoList.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;

@Entity
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String title;
    private String description;
    private String status;
    @JsonProperty("dueDate")
    private String dueDate;

    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "userid")
    private User user;

    public Task() {
        id = 0;
        title = "";
        description = "";
        status = "";
        dueDate = "";
        user = null;
    }

    public Task(int id, String title, String description, String status, String dueDate, User user) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.status = status;
        this.dueDate = dueDate;
        this.user = user;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getDueDate() {
        return dueDate;
    }

    public void setDueDate(String dueDate){
        this.dueDate = dueDate;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}
